client/core/core.go

```diff
@@ -2904,6 +2906,61 @@ func (c *Core) tipChange(assetID uint32, nodeErr error) {
 	c.updateBalances(counts)
 }
 
+func (c *Core) PromptShutdown() bool {
+	c.connMtx.Lock()
+	defer c.connMtx.Unlock()
+	ok := true
+	for _, dc := range c.conns {
+		if dc.hasActiveOrders() {
+			ok = false
+			break
+		}
+	}
+	if !ok {
+		fmt.Print("You have active orders. Shutting down now may result in failed swaps and account penalization. " +
+			"Do you want to quit anyway? ('y' to quit, enter to abort):")
+		scanner := bufio.NewScanner(os.Stdin)
+		scan := make(chan bool)
+		go func() {
+			scan <- scanner.Scan()
+		}()
+		select {
+		case <-c.ctx.Done():
+			return ok
+		case <-scan:
+		}
+
+		promptResp := scanner.Text()
+		err := scanner.Err()
+		if err != nil {
+			fmt.Printf("input failed: %v", err)
+			return ok
+		}
+
+		switch promptResp {
+		case "y", "yes":
+			for assetID := range c.User().Assets {
+				wallet, found := c.wallet(assetID)
+				if found && wallet.connected() {
+					if err := wallet.Lock(); err != nil {
+						fmt.Println(err)
+						return ok
+					}
```

Comment:
I think shutdown should continue regardless of lock error and not return here. Of course log it, but this shouldn't keep the user from stopping the app.